### PR TITLE
Feat-828/Simple Text File

### PR DESCRIPTION
**Pull request by lukman7788**
# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Create config from simple text file (#828 )

## How did you implement / how did you fix it  
1.  add feature to run config from simple text file
2. add feature to create config from simple text file

## How to test  
1. `/bin/run_dev --text ./src/components/config/__tests__/textfile`
2. `npm run test`

valid url
image

invalid url
image

### CODE REVIEW

File: src/components/config/parse-text.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import { Config } from '../../interfaces/config'
+import { DEFAULT_CONFIG_INTERVAL } from '.'
+import { DEFAULT_THRESHOLD } from '../../looper'
+import { Probe } from '../../interfaces/probe'
+
+const generateProbesFromText = (config: any): Probe[] => {
+  const probes = []
+  const urls = config.split(/\r?\n/)
+  for (const [, url] of urls.entries()) {
```
Comment (nicnocquee) [inline src/components/config/parse-text.ts:33]: can we validate the `urls` variable using `Yup.array().of(Yup.string().url())`? To make sure the urls are all valid.

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: src/components/config/parse-text.ts (hunk 1 above)
Comment (lukman7788) [reply src/components/config/parse-text.ts]: updated using existing url checker (`import { isValidURL } from '../../utils/is-valid-url'`)

File: src/components/config/parse-text.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                assertion: 'response.time > 2000',
+                message: 'Response time is more than 2000ms',
+              },
+            ],
+          },
+        ]
+      }
+      // Ignoring new line from url validating checker
+      else if (url.length > 0) {
+        throw new Error('URL invalid')
```
Comment (nicnocquee) [inline src/components/config/parse-text.ts:92]: to make it clearer, `The URL ${url} is not a valid URL`

File: src/components/config/parse-text.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      }
+    }
+
+    return { probes }
+  } catch {
+    throw new Error('Your Text file contains an invalid format !')
+  }
```
Comment (nicnocquee) [inline src/components/config/parse-text.ts:99]: Do we need to catch it here if it's gonna throw another Error again?

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: src/components/config/parse-text.ts (hunk 2 above)
Comment (lukman7788) [reply src/components/config/parse-text.ts]: updated  @nicnocquee

File: src/components/config/parse-text.ts (hunk 3 above)
Comment (lukman7788) [reply src/components/config/parse-text.ts]: try catch removed @nicnocquee

**Review by nicnocquee — APPROVED** [review]

**Pull request merged**